gramps/gen/proxy/proxybase.py

```diff
@@ -163,44 +189,69 @@ def include_something(self, handle, obj=None):
         return obj.include()
 
     # Define default predicates for each object type
+    def include_person(self, handle: PersonHandle) -> bool:
+        raise NotImplementedError
+
+    def include_family(self, handle: FamilyHandle) -> bool:
+        raise NotImplementedError
+
+    def include_event(self, handle: EventHandle) -> bool:
+        raise NotImplementedError
+
+    def include_source(self, handle: SourceHandle) -> bool:
+        raise NotImplementedError
+
+    def include_citation(self, handle: CitationHandle) -> bool:
+        raise NotImplementedError
+
+    def include_place(self, handle: PlaceHandle) -> bool:
+        raise NotImplementedError
+
+    def include_media(self, handle: MediaHandle) -> bool:
+        raise NotImplementedError
 
-    include_person = include_family = include_event = include_source = (
-        include_citation
-    ) = include_place = include_media = include_repository = include_note = (
-        include_tag
-    ) = None
+    def include_repository(self, handle: RepositoryHandle) -> bool:
+        raise NotImplementedError
```

Comment:
I was not 100% sure if this change is correct. Any python experts care to comment?
I need to give the predicates the correct type

Reply:
answering my own question, it's not. It alters behaviour in the `gfilter` method which has this line:
`if predicate is not None and obj is not None:`